temporian/implementation/numpy/data/tests/test_df_to_event.py

```diff
@@ -0,0 +1,189 @@
+from temporian.implementation.numpy.data.sampling import NumpySampling
+
+
+class DataFrameToEventTest(absltest.TestCase):
+    def test_df_to_numpy_event(self) -> None:
+        df = pd.DataFrame(
+            [
+                [666964, 1.0, 740.0],
+                [666964, 2.0, 508.0],
+                [574016, 3.0, 573.0],
+            ],
+            columns=["product_id", "timestamp", "costs"],
+        )
+
+        numpy_sampling = NumpySampling(
+            data={
+                (666964,): np.array([1.0, 2.0]),
+                (574016,): np.array([3.0]),
+            },
+            index=["product_id"],
+        )
+
+        expected_numpy_event = NumpyEvent(
+            data={
+                (666964,): [
+                    NumpyFeature(data=np.array([740.0, 508.0]), name="costs")
+                ],
+                (574016,): [NumpyFeature(data=np.array([573.0]), name="costs")],
+            },
+            sampling=numpy_sampling,
+        )
+
+        numpy_event = NumpyEvent.from_dataframe(
+            df, index_names=["product_id"], timestamp_column="timestamp"
+        )
+
+        # validate
+        self.assertTrue(numpy_event == expected_numpy_event)
+
+    def test_df_to_numpy_event_string(self) -> None:
+        df = pd.DataFrame(
+            [
+                [666964, 1.0, "740.0"],
+                [666964, 2.0, "508.0"],
+                [574016, 3.0, "573.0"],
+            ],
+            columns=["product_id", "timestamp", "costs"],
+        )
+
+        self.assertRaisesRegex(
+            ValueError,
+            "Unsupported dtype",
+            NumpyEvent.from_dataframe,
+            df,
+            ["product_id"],
+        )
+
+    def test_df_to_numpy_event_missing_values(self) -> None:
+        df = pd.DataFrame(
+            [
+                [666964, 1.0, 740.0],
+                [666964, 2.0],
+                [574016, 3.0, 573.0],
+            ],
+            columns=["product_id", "timestamp", "costs"],
+        )
+
+        numpy_sampling = NumpySampling(
+            data={
+                (666964,): np.array([1.0, 2.0]),
+                (574016,): np.array([3.0]),
+            },
+            index=["product_id"],
+        )
+
+        expected_numpy_event = NumpyEvent(
+            data={
+                (666964,): [
+                    NumpyFeature(data=np.array([740.0, np.nan]), name="costs")
+                ],
+                (574016,): [NumpyFeature(data=np.array([573.0]), name="costs")],
+            },
+            sampling=numpy_sampling,
+        )
+
+        numpy_event = NumpyEvent.from_dataframe(
+            df, index_names=["product_id"], timestamp_column="timestamp"
+        )
+
+        # validate
+        self.assertTrue(numpy_event == expected_numpy_event)
+
+    def test_df_to_numpy_event_datetime_index(self) -> None:
+        df = pd.DataFrame(
+            [
+                [666964, np.datetime64("2022-01-01"), 740.0],
```

Comment:
Can you test the other supported datetime object, and ensure that the timestamp is as expected?